ACPs/224-dynamic-gas-limit-in-subnet-evm/README.md

```diff
@@ -0,0 +1,233 @@
+
+```json
+{
+  ...
+  "acp224Timestamp": ...,
+  "acp224FeeConfig": {
+    "targetGas": ...,
+    "minGasPrice": ...,
+    "maxCapacityFactor": ...,
+    "timeToDouble": ...,
+    "targetBlockRate": ...,
+    "minBlockGasCost": ...,
+    "maxBlockGasCost": ...,
+    "blockGasCostStep": ...
+  }
+}
+
+```
+
+### New ACP176 Fee Manager Precompile
+
+A new fee manager precompile will be required to dynamically changing the parameters. The precompile will be similar to the existing fee manager precompile. The solidity interface will be as follows:
+
+```solidity
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+import "./IAllowList.sol";
+
+interface IACP176FeeManager is IAllowList {
+  struct ACP176FeeConfig {
+    uint256 targetGas;
+    uint256 minGasPrice;
+    uint256 maxCapacityFactor;
+    uint256 timeToDouble;
+    uint256 targetBlockRate;
+    uint256 minBlockGasCost;
+    uint256 maxBlockGasCost;
+    uint256 blockGasCostStep;
+  }
+  event FeeConfigChanged(address indexed sender, ACP176FeeConfig oldFeeConfig, ACP176FeeConfig newFeeConfig);
+
+  // Set fee config fields to contract storage
+  function setFeeConfig(
+    uint256 targetGas;
+    uint256 minGasPrice;
+    uint256 maxCapacityFactor;
+    uint256 timeToDouble;
+    uint256 targetBlockRate;
+    uint256 minBlockGasCost;
+    uint256 maxBlockGasCost;
+    uint256 blockGasCostStep;
+  ) external;
+
+  // Get fee config from the contract storage
+  function getFeeConfig()
+    external
+    view
+    returns (
+      uint256 targetGas;
+      uint256 minGasPrice;
+      uint256 maxCapacityFactor;
+      uint256 timeToDouble;
+      uint256 targetBlockRate;
+      uint256 minBlockGasCost;
+      uint256 maxBlockGasCost;
+      uint256 blockGasCostStep;
+    );
+
+  // Get the last block number changed the fee config from the contract storage
+  function getFeeConfigLastChangedAt() external view returns (uint256 blockNumber);
+}
+```
+
+The precompile will have a `targetGas` parameter that is used to set the target gas consumed per second for ACP-176. This will override the `gas-target` in node config. If the new fee manager precompile is not activated, validators can keep using `gas-target` in node config to set the target gas consumed per second (as in C-Chain).
+
+```mermaid
+flowchart TD
+    A[ACP-224 activated] --> B{Is ACP176FeeManager precompile active?}
+
+    B -- Yes --> C[Use targetGas from precompile storage]
+
+    B -- No --> D{Is gas-target set in node config?}
+    D -- Yes --> E[Use gas-target from node config]
+
+    D -- No --> F{Is targetGas set in chainConfig.acp176FeeConfig?}
```

Comment:
What's the "node config" value here? If you mean the value set for the node for the C-Chain, I don't think we should be re-using it here, and could probably simplify this by just looking at the chain config.

Reply:
the node config is `target-gas` flag in per-node configuration (as the one in c-chain). Are you implying to completely removing the node configuration and just depend on genesis chain config (or whatever the default is)? I was thinking to keep this mechanism in case some L1s (no owned subnets) want to leave this to their validators (like C-chain).

Reply:
Definitely agree that we want to keep this mechanism to allow for validators to set their preference, but we should require it to be specified on a per-chain basis for each node. I don't think there should be any `target-gas` node config that is applied to all EVM chains run by a given node. Each node should specify the `target-gas` for each chain individually, including the C-Chain.

Reply:
ahh I see what you mean, no this should never be the case (and it would even more difficult to do so). What I mean by "node config" is a config that can be changed per node per chain. This will be equivalent to chainConfigs under `~/.avalanchego/configs/<chainID>/config.json`. I specifically wanted to avoid `chain config` because we also have a `chain config` in genesis json.